tests/serial/mapping-greedy-kernel/helpers.cpp

```diff
@@ -0,0 +1,182 @@
+  ids.emplace_back(interface.setMeshVertex(meshTwoID, Eigen::Vector3d{3.5, 0.5, z}));
+  return ids;
+}
+
+
+void testGreedyMappingDirection1(const std::string configFile, const TestContext &context)
+{
+  using Eigen::Vector3d;
+
+  std::vector<double> values;
+  for (unsigned int i = 0; i < 12; ++i)
+    values.emplace_back(std::pow(i + 1, 2));
+
+  double expectedValues[3] = {1.0000000000000002, 2.1879131472090689, 22.859664317930537};
+
+  if (context.isNamed("SolverOne")) {
+    precice::Participant interface("SolverOne", configFile, 0, 1);
+    // namespace is required because we are outside the fixture
+    auto meshOneID = "MeshOne";
+
+    // Setup mesh one.
+    std::vector<int> ids = generateMeshOne(interface, meshOneID);
+
+    // Initialize, thus sending the mesh.
+    interface.initialize();
+    double maxDt = interface.getMaxTimeStepSize();
+    BOOST_TEST(interface.isCouplingOngoing(), "Sending participant should have to advance once!");
+
+    // Write the data to be send.
+    auto dataAID = "DataOne";
+    BOOST_TEST(!interface.requiresGradientDataFor(meshOneID, dataAID));
+    interface.writeData(meshOneID, dataAID, ids, values);
+
+    // Advance, thus send the data to the receiving partner.
+    interface.advance(maxDt);
+    BOOST_TEST(!interface.isCouplingOngoing(), "Sending participant should have to advance once!");
+    interface.finalize();
+
+  } else {
+    BOOST_TEST(context.isNamed("SolverTwo"));
+    precice::Participant interface("SolverTwo", configFile, 0, 1);
+    // namespace is required because we are outside the fixture
+    auto meshTwoID = "MeshTwo";
+
+    // Setup receiving mesh.
+    std::vector<int> ids = generateMeshTwo(interface, meshTwoID);
+
+    // Initialize, thus receive the data and map.
+    interface.initialize();
+    double maxDt = interface.getMaxTimeStepSize();
+    BOOST_TEST(interface.isCouplingOngoing(), "Receiving participant should have to advance once!");
+
+    // Read the mapped data from the mesh.
+    auto dataAID = "DataOne";
+    BOOST_TEST(!interface.requiresGradientDataFor(meshTwoID, dataAID));
+
+    double values[3];
+    interface.readData(meshTwoID, dataAID, ids, maxDt, values);
+
+    for (int i = 0; i < 3; i++) {
+      BOOST_TEST(values[i] == expectedValues[i], boost::test_tools::tolerance(1e-7));
+    }
+
+
+    // Verify that there is only one time step necessary.
+    interface.advance(maxDt);
+    BOOST_TEST(!interface.isCouplingOngoing(), "Receiving participant should have to advance once!");
+    interface.finalize();
+  }
+}
+
+
+void testGreedyMappingDirection2(const std::string configFile, const TestContext &context)
+{
+  using Eigen::Vector3d;
+
+  auto meshOneID = "MeshOne";
+  auto meshTwoID = "MeshTwo";
+  auto dataAID = "DataOne";
+
+  std::vector<double> values;
+  for (unsigned int i = 0; i < 3; ++i)
+    values.emplace_back(std::pow(i + 1, 2));
+
+  double expectedValues[12] = {
```

Comment:
I assume you put these values as they were generated by the algorithm, right? It would be good to also have some more simple test cases, for which you know the results analytically, e.g., if an input vertex is also in the output set, then input should be equal to output, or if we have a linear function between two vertices and evaluate in the middle, there should be a reasonable guess on the result. We could also consider to perform regression testing against the `rbf-direct` solver. Maybe you can simply re-use some of the existing solver tests we have https://github.com/precice/precice/blob/develop/src/mapping/tests/GinkgoRadialBasisFctSolverTest.cpp

This makes debugging more easy in case test cases fail.

Reply:
The values were generated by rbf-global-direct. The greedy algorithms should interpolate the function exactly the same way as rbf-global-direct when including every input vertex.